src/components/badge/badge.tsx

```diff
@@ -58,6 +59,7 @@ export type EuiBadgeProps = {
    * Props passed to the close button.
    */
   closeButtonProps?: Partial<PropsOf<EuiIcon>>;
+  theme?: 'light' | 'dark';
```

Comment:
I assume this theming example is just some boilerplate to show the possibility. Since this would happen in every one of our files, I assume we'd not need to redeclare it in every file?

Also, how do you think we'd handle a dynamically user defined (say through a UI and stored in kibana somewhere) third theme?

Reply:
So I think I understand how this might work after I figured out the sass import bit from above. We'd essentially have a Sass template that redeclared all the variables (dynamically, rather than from set files) and then figure a way to import it anytime the styled-jsx stuff was called.

So rather than do ...

```
  const { className: themeClass, styles: themeStyles } = css.resolve`
    @import 'badge';
  `;

  const { className: darkThemeClass, styles: darkThemeStyles } = css.resolve`
    @import '../../themes/eui/eui_colors_dark';
    @import 'badge';
  `;
```

we'd do...

```
  const { className: themeClass, styles: themeStyles } = css.resolve`
    ${themeVarsFromContext}
    @import 'badge';
  `;
```

and then do the same when the JSX tag is called.
```
      <style jsx>{`
        ${themeVarsFromContext}
        @import '../../global_styling/context';
        .euiAvatar {
          background-image: ${imageUrl ? `url(${imageUrl})` : 'none'};
          background-color: ${assignedColor};
          @debug ${assignedColor};
          color: chooseLightOrDarkText(
            $euiColorPrimary,
            $euiColorGhost,
            $euiColorInk
          );
          border: solid 1px $euiColorPrimary;
        }
      `}</style>
```

The curious bit is... can we do this automatically somehow without having to declare this stuff everywhere? Like a setting in style-jsx itself that's the equiv of passing env_vars?

I'm talking random now so I will stop!

Reply:
> I assume this theming example is just some boilerplate to show the possibility 

Yes. I imagine a top-level service or React context layer that would provide a way for all our components to just "get" the active theme.

> can we do this automatically somehow without having to declare this stuff everywhere?

Possibly. We have full control over the Sass compilation step (it's a plugin) so we may be able to just provide all of our Sass globals at compile time and without explicit `@import`s all over.